File: app/src/main/java/com/apkakisan/myapplication/HomeActivity.java
package com.apkakisan.myapplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.SearchView;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.navigation.NavigationBarView;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HomeActivity extends AppCompatActivity implements CommoditiesAdapter.OnCommodityListener{
    RecyclerView recyclerView;
    ProgressBar progressBar;
    LinearLayoutManager layoutManager;
    CommoditiesAdapter adapter;
    List<CommodityItem> commodityItemList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);


        recyclerView = findViewById(R.id.recycler_view);
        progressBar = findViewById(R.id.progressBar);
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        adapter = new CommoditiesAdapter(commodityItemList, this);
        recyclerView.setAdapter(adapter);

        SearchView searchView = findViewById(R.id.searchView);


        fetchCommodities();

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        bottomNavigationView.setSelectedItemId(R.id.home);

        bottomNavigationView.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()) {
                    case R.id.home:
//                        Toast.makeText(MainContainer.this, "HOME SELECTED", Toast.LENGTH_SHORT).show();
                        return true;
                    case R.id.orders:
//                        Toast.makeText(MainContainer.this, "ORDER SELECTED", Toast.LENGTH_SHORT).show();
                        startActivity(new Intent(getApplicationContext(), OrdersActivity.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.notifications:
//                        Toast.makeText(MainContainer.this, "NOTIFICATION SELECTED", Toast.LENGTH_SHORT).show();
                        startActivity(new Intent(getApplicationContext(), NotificationsActivity.class));
                        overridePendingTransition(0,0);
                        return true;
                    case R.id.profile:
//                        Toast.makeText(MainContainer.this, "PROFILE SELECTED", Toast.LENGTH_SHORT).show();
                        startActivity(new Intent(getApplicationContext(), ProfileActivity.class));
                        overridePendingTransition(0,0);
                        return true;
                }
                return false;
            }
        });


        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String s) {
//                Toast.makeText(HomeActivity.this, "text is- "+s, Toast.LENGTH_SHORT).show();
//                System.out.println("adapterrrrrr--> "+adapter);
                filter(s);
                return true;
            }
        });

    }

    public void filter(String s) {
        List<CommodityItem> filteredList = new ArrayList<>();
        for(CommodityItem item : commodityItemList) {
            if(item.getCommodity_title().toLowerCase().contains(s.toLowerCase())) {
                filteredList.add(item);
            }
        }
        adapter.filterList(filteredList);
    }


    private void fetchCommodities() {

        progressBar.setVisibility(View.VISIBLE);

        RetrofitClient.getRetrofitClient().getCommodity_title().enqueue(new Callback<List<CommodityItem>>() {
            @Override
            public void onResponse(Call<List<CommodityItem>> call, Response<List<CommodityItem>> response) {
                
                if(response.isSuccessful() && response.body() != null) {
//                    Toast.makeText(HomeActivity.this, "Response body: " + response.body(), Toast.LENGTH_SHORT).show();
//                    System.out.println("here---->>>> "+response.body());
                    commodityItemList.addAll(response.body());
                    adapter.notifyDataSetChanged();
                    progressBar.setVisibility(View.GONE);
                }
            }

            @Override
            public void onFailure(Call<List<CommodityItem>> call, Throwable t) {
                progressBar.setVisibility(View.GONE);
                Toast.makeText(HomeActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public void onCommodityClick(int position) {
        commodityItemList.get(position);
        Intent intent = new Intent(HomeActivity.this, CreateOrder.class);
        intent.putExtra("commodity_name", commodityItemList.get(position).getCommodity_title());
        intent.putExtra("commodity_apka_kisan_price", commodityItemList.get(position).getCommodity_apkakisan_price());

        startActivity(intent);
    }
}

File: app/src/main/java/com/apkakisan/myapplication/order/HomeActivity.kt
package com.apkakisan.myapplication.order

import androidx.recyclerview.widget.RecyclerView
import android.widget.ProgressBar
import androidx.recyclerview.widget.LinearLayoutManager
import com.apkakisan.myapplication.network.responses.Commodity
import android.os.Bundle
import com.apkakisan.myapplication.R
import android.content.Intent
import android.view.View
import com.apkakisan.myapplication.network.RetrofitClient
import android.widget.Toast
import androidx.appcompat.widget.SearchView
import com.apkakisan.myapplication.BaseActivity
import com.apkakisan.myapplication.helpers.DefaultItemDecorator
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.ArrayList

class HomeActivity : BaseActivity() {
    private lateinit var recyclerView: RecyclerView
    private lateinit var progressBar: ProgressBar

    private var commodityItemList: MutableList<Commodity> = ArrayList()

    private lateinit var adapter: CommoditiesAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        recyclerView = findViewById(R.id.recycler_view)
        progressBar = findViewById(R.id.progressBar)

        val searchView = findViewById<SearchView>(R.id.searchView)

        bottomNavigation = findViewById(R.id.layoutBottom)
        bottomNavigation.selectedItemId = R.id.home
        setupBottomNavigation()

        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(s: String): Boolean {
                return false
            }

            override fun onQueryTextChange(s: String): Boolean {
                filter(s)
                return true
            }
        })

        adapterProcess()
        fetchCommodities()
    }

    private fun adapterProcess() {
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.addItemDecoration(DefaultItemDecorator(48, 48))
        adapter = CommoditiesAdapter(
            commodityItemList,
            onClick = { commodity: Commodity, _: View, _: Int ->
                startActivity(Intent(this, CreateOrderActivity::class.java).apply {
                    putExtra(CreateOrderActivity.TITLE, commodity.title)
                    putExtra(CreateOrderActivity.MANDI_PRICE, commodity.mandiPrice)
                    putExtra(CreateOrderActivity.APKAKISAN_PRICE, commodity.apkakisanPrice)
                })
            }
        )
        recyclerView.adapter = adapter
    }

    private fun fetchCommodities() {
        progressBar.visibility = View.VISIBLE
        RetrofitClient.getRetrofitClient().commodity_title.enqueue(
            object : Callback<List<Commodity>?> {
                override fun onResponse(
                    call: Call<List<Commodity>?>,
                    response: Response<List<Commodity>?>
                ) {
                    if (response.isSuccessful && response.body() != null) {
                        commodityItemList.addAll(response.body() ?: listOf())
                        adapter.notifyDataSetChanged()
                        progressBar.visibility = View.GONE
                    }
                }

                override fun onFailure(call: Call<List<Commodity>?>, t: Throwable) {
                    progressBar.visibility = View.GONE
                    Toast.makeText(this@HomeActivity, "Error: " + t.message, Toast.LENGTH_SHORT)
                        .show()
                }
            })
    }

    fun filter(s: String) {
        val filteredList: MutableList<Commodity> = ArrayList()
        for (item in commodityItemList) {
            if (item.title.lowercase().contains(s.lowercase())) {
                filteredList.add(item)
            }
        }
        adapter.filterList(filteredList)
    }
}